app/controllers/vendors_controller.rb

```diff
@@ -0,0 +1,54 @@
+class VendorsController < ApplicationController
+
+  def index
+    @vendors = Vendor.order(vendor_name: :asc)
+    render :index
+  end
+
+  def show
+    @vendors = Vendor.where(id: params[:id])
+    render :single_vendor_view
+  end
+
+  def new
+    @new_vendor = Vendor.new
+    @market = Market.where(id: params[:id]).first
+  end
+
+  def create
+    @new_vendor = Vendor.new(new_vendor_create_params[:vendor])
+    @mark_id = @new_vendor.market_id
```

Comment:
This seems to be an overuse of instance variables. You only need to create an instance variable when it is being accessed by the view. Otherwise, it can just be a local variable (mark_id instead of @mark_id).